src/resmom/linux/alps.c

```diff
@@ -323,6 +393,31 @@ new_alps_req(void)
 	requestCurr = 0;
 }
 
+/**
+ * @brief
+ * Start a new ALPS request for KNL.
+ *
+ * If need be, allocate a buffer.
+ * @retval 1 if buffer allocation failed.
+ * @retval 0 if success.
+ */
+static int
+new_alps_req_KNL(void)
+{
+	int ret_val = 0;
+
+	if (requestBuffer_knl == NULL) {
+		requestSize_knl = UTIL_BUFFER_LEN;
+		requestBuffer_knl = (char*)malloc(UTIL_BUFFER_LEN);
+	}
+	if (!requestBuffer_knl) {
+		snprintf(log_buffer, sizeof(log_buffer), "Memory allocation for XML request buffer failed.");
+		log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_NODE, LOG_ERR, __func__, log_buffer);
+		ret_val = 1;
+	}
+	return ret_val;
+}
```

Comment:
We can do little code optimization, by avoiding the use of 'ret_val' and inlining the second if statement with malloc.

Reply:
Done!